src/shared/filesystemUtilities.ts

```diff
@@ -65,14 +70,16 @@ export async function findFileInParentPaths(searchFolder: string, fileToFind: st
     return findFileInParentPaths(parentPath, fileToFind)
 }
 
-export const getTempDirPath = (prefix: string = 'vsctk') => {
-    return path.join(
-        os.type() === 'Darwin' ? '/tmp' : os.tmpdir(),
-        prefix
-    )
-}
-
 const _mkdtemp = promisify(fs.mkdtemp)
-export const  mkdtemp = async (prefix?: string) => {
-    return _mkdtemp(getTempDirPath(prefix))
+export const  mkdtemp = async (...relativePathParts: string[]) => {
```

Comment:
Thinking out loud here, I wonder if the function should be named something like `makeTemporaryToolkitFolder` indicating that it is specifically a toolkit related temp folder.

Reply:
That would not be terrible, although generally I'm a fan of standard names over having to think of a special name for the same concept in each domain.

e.g. Concept `name`

1. Standard property name:
 a. `Person` with `name` `Person { private name: string }`
 b. `Pet` with `name` `Pet { private name: string }`

2. Different property name per domain:
 a. `Person` with `name` `Person { private personName: string }`
 b. `Pet` with `name` `Pet { private petName: string }`

Another unrelated undesirable side effect of option 2 is that it makes generic functions yucky at best. Option 2 makes me sad.

For those familiar with `fs`, they would instinctively think of the standard name and then have to mentally and likely physically search for the corresponding overridden concept in each domain.

Reply:
The main motivation would be to ensure there isn't an expectation that the standard named function has the same standard behavior. Your choice though.

Reply:
+1. To me, I'd expect a function with a standard name to have the standard behavior for that name. Since this function does not, we should have a more specific name.

Reply:
Unfortunately I started from a different base and neglected to squash, but I previously did a rebase hence the diff is correct. Consolidating the commit messages is part of squash which will happen when we smash and merge right?

Reply:
All issues addressed here: https://github.com/aws/aws-toolkit-vscode/pull/363/files

Reply:
✔️